libs/wingsdk/src/core/resource.ts

```diff
@@ -154,26 +154,30 @@ export abstract class Resource
       if (field.startsWith("this.")) {
         const key = field.substring(5);
         const obj: Resource = (this as any)[key];
-        if (!obj) {
+        if (obj === undefined) {
           throw new Error(
             `Resource ${this.node.path} does not have field ${key}`
           );
         }
-        if (!("_bind" in obj)) {
-          throw new Error(
-            `Resource ${this.node.path} field ${key} does not have a bind method`
-          );
-        }
-        obj._bind(host, resources[field]);
 
-        // add connection metadata
-        for (const op of resources[field]) {
-          Resource.addConnection({
-            from: host,
-            to: obj,
-            relationship: op,
-          });
+        // if this is a bindable object, call `_bind()` and add a connection for each operation.
+        if (typeof obj === "object" && "_bind" in obj) {
```

Comment:
Back in the days of the original polycons RFC, I remember thinking that capturing (or binding) should probably have an interface associated with it. IBindable or whatever.
 
Resources would obviously implement that interface, but then primitives would also have a similar Primitive class that implements it with plain old serialization. Then the compiler could emit calls to `Primitive.binding("my object")`). I figured this would be necessary if we wanted to be able to bind arbitrary classes.

Does that path make sense, and/or is it an unnecesarry complication?